sympy/solvers/ode/single.py

```diff
@@ -1889,5 +1889,87 @@ def _get_general_solution(self, *, simplify_flag: bool = True):
         return fsol
 
 
+class Hypergeometric2nd(SingleODESolver):
+    r"""
+    Solves 2nd order linear differential equations.
+
+    It computes special function solutions which can be expressed using the
+    2F1, 1F1 or 0F1 hypergeometric functions.
+
+    .. math:: y'' + A(x) y' + B(x) y = 0\text{,}
+
+    where `A'/2 + A^2/4 - B`  is a rational function of x.
+
+    These kinds of differential equations have solution of non-Liouvillian form.
+
+    Given linear ODE can be obtained from 2F1 given by
+
+    .. math:: (x^2 - x) y'' + ((a + b + 1) x - c) y' + b a y = 0\text{,}
```

Comment:
I think this would make a better example for the doctest.

Reply:
Yes, I tried that but only classify_ode works fine for this. dsolve returns "This equation cannot be solved by hypergeometric"

Reply:
Oh well that's a bug. Can you open an issue? Maybe CC @RituRajSingh878 in the issue